### PR TITLE
Fixes #18141 - Move environment override to pagelet

### PR DESCRIPTION
**Pull request by ShimShtein**
No need to use deface in order to add fields to host's main tab, or adding additional tabs to host edit form.

### CODE REVIEW

**Comment by ShimShtein** [comment]
This is a proper fix instead of #6556

File: app/overrides/add_activation_keys_input.rb [hunk 1]
```diff
@@ -13,10 +13,6 @@
                      :insert_before => 'erb[loud]:contains("hostgroup_puppet_environment_field")',
                      :partial => 'overrides/activation_keys/host_environment_select')
 
-Deface::Override.new(:virtual_path => "hosts/_form",
-                     :name => "hosts_update_environments_select",
-                     :insert_after => 'erb[loud]:contains("Deploy on")',
-                     :partial => 'overrides/activation_keys/host_environment_select')
```
Comment (jlsherrill) [inline app/overrides/add_activation_keys_input.rb]: This results in an extra blank line which is causing rubocop to fail hounci--

File: config/initializers/pagelets.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# initialize katello related pagelets
+mgr = Pagelets::Manager.new "hosts/_form"
+
+mgr.add_pagelet :main_tab_fields,
+  :partial => "overrides/activation_keys/host_environment_select",
+  :priority => 200
```
Comment (jlsherrill) [inline config/initializers/pagelets.rb]: this places the lifecycle end environment selector below the puppet environment selector, but I think it should be above, thoughts on lowering the priority?

**Comment by jlsherrill** [comment]
Thanks for doing this BTW!

**Comment by ShimShtein** [comment]
Lowered the priority to 80 (puppet has 100).
Removed the empty line.

**Comment by jlsherrill** [comment]
@ShimShtein looks like a unt test error in jenkins: http://ci.theforeman.org/job/test_katello_pull_request/11016/slave=fast/testReport/junit/(root)/HostgroupsControllerTest/test_new/

**Comment by ShimShtein** [comment]
Now it should work for hostgroups too.

**Comment by jlsherrill** [comment]
everything looks good, i'm ready to merge, but @ShimShtein what did you mean by 'it should work for hostgroups too' ?  

Looks like the hostgroups form is still using deface in your commit.

**Comment by ShimShtein** [comment]
Katello uses the same template for hosts and hostgroups, but only hosts have the pagelets feature. That was the reason the tests were failing previously - pagelets changed the notion a bit, and renamed the local `f` to something a bit more meaningful (`form`).
Now the code should work for both pagelets and deface use case.

And yes, I know I should [file a refactor](http://projects.theforeman.org/issues/18350) for hostgroups form too.

**Comment by jlsherrill** [comment]
ACK thanks @ShimShtein !

**Pull request merged**